frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx

```diff
@@ -0,0 +1,257 @@
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDeposit.toInterfaceModel(depositToDelete),
+      });
+      if (!!updatedLease?.id) {
+        setDepositToDelete(undefined);
+        setDeleteModalWarning(false);
+        setLease(updatedLease);
+      }
+    }
+  };
+
+  const onDeleteDepositReturnConfirmed = async () => {
+    if (lease && lease.id && lease.rowVersion && depositReturnToDelete) {
+      const updatedLease = await removeLeaseDepositReturn({
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDepositReturn.toInterfaceModel(depositReturnToDelete),
+      });
+      if (!!updatedLease?.id) {
+        setDepositReturnToDelete(undefined);
+        setDeleteReturnModalWarning(false);
+        setLease(updatedLease);
+      }
+    }
+  };
+
+  /**
+   * Send the save request (either an update or an add). Use the response to update the parent lease.
+   * @param depositForm
+   */
+  const onSaveDeposit = async (depositForm: FormLeaseDeposit) => {
+    console.log('saving deposit!', depositForm);
+    if (lease && lease.id && lease.rowVersion) {
+      let request: IParentConcurrencyGuard<ILeaseSecurityDeposit> = {
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDeposit.toInterfaceModel(depositForm),
+      };
+      const updatedLease = await updateLeaseDeposit(request);
+      if (!!updatedLease?.id) {
+        setLease(updatedLease);
+        setEditDepositValue(FormLeaseDeposit.createEmpty());
+        setShowEditModal(false);
+      }
+    } else {
+      console.log('Lease information incomplete');
+    }
+  };
+
+  const onEditReturnDeposit = (id: number) => {
+    var deposit = depositReturns.find(x => x.id === id);
+    if (deposit) {
+      var parentDeposit = securityDeposits.find(x => x.id === deposit?.parentDepositId);
+      if (parentDeposit) {
+        setEditReturnValue(FormLeaseDepositReturn.createFromModel(deposit, parentDeposit));
+        setShowReturnEditModal(true);
+      } else {
+        console.log('Parent deposit incomplete');
+      }
+    }
+  };
+
+  const onDeleteDepositReturn = (id: number) => {
+    var deposit = depositReturns.find(x => x.id === id);
+    if (deposit) {
+      var parentDeposit = securityDeposits.find(x => x.id === deposit?.parentDepositId);
+      if (parentDeposit) {
+        setDepositReturnToDelete(FormLeaseDepositReturn.createFromModel(deposit, parentDeposit));
+        setDeleteReturnModalWarning(true);
+      } else {
+        console.log('Parent deposit incomplete');
+      }
+    }
+  };
+
+  /**
+   * Send the save request (either an update or an add). Use the response to update the parent lease.
+   * @param returnDepositForm
+   */
+  const onSaveReturnDeposit = async (returnDepositForm: FormLeaseDepositReturn) => {
+    console.log('saving return deposit!', returnDepositForm);
```

Comment:
I'm not against adding logging like this to our backend and frontend but I don't think we want to do it via console logs. Will check if we have a tech debt item for logging on the FE (via a library).

Reply:
I agree, it is here only for debugging purposes while the tests are completed